File: t.rb
# https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/metrics/cyclomatic_complexity.rb
# https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/metrics/perceived_complexity.rb
def if_elseif_test
  car = "Patriot"

  manufacturer =
    if car == "Focus"
      "Ford"
    elsif car == "Navigator"
      "Lincoln"
    elsif car == "Camry"
      "Toyota"
    elsif car == "Civic"
      "Honda"
    elsif car == "Patriot"
      "Jeep"
    elsif car == "Jetta"
      "VW"
    elsif car == "Ceyene"
      "Porsche"
    elsif car == "Outback"
      "Subaru"
    elsif car == "520i"
      "BMW"
    elsif car == "Tundra"
      "Nissan"
    else
      "Unknown"
    end

  manufacturer
end

def case_when_test
	car = "Patriot"

	manufacturer = case car
	   when "Focus" then "Ford"
	   when "Navigator" then "Lincoln"
	   when "Camry" then "Toyota"
	   when "Civic" then "Honda"
	   when "Patriot" then "Jeep"
	   when "Jetta" then "VW"
	   when "Ceyene" then "Porsche"
	   when "Outback" then "Subaru"
	   when "520i" then "BMW"
	   when "Tundra" then "Nissan"
	   else "Unknown"
	end
end

File: t.py
def match_case_test():
    car = "Patriot"
    
    match car:
        case "Focus":
            manufacturer = "Ford"
        case "Navigator":
            manufacturer = "Lincoln"
        case "Camry":
            manufacturer = "Toyota"
        case "Civic":
            manufacturer = "Honda"
        case "Patriot":
            manufacturer = "Jeep"
        case "Jetta":
            manufacturer = "VW"
        case "Ceyene":
            manufacturer = "Porsche"
        case "Outback":
            manufacturer = "Subaru"
        case "520i":
            manufacturer = "BMW"
        case "Tundra":
            manufacturer = "Nissan"
        case "Tundra2":
            manufacturer = "Nissan2"
        case "Tundra3":
            manufacturer = "Nissan3"
        case "Tundra4":
            manufacturer = "Nissan4"
        case "Tundra5":
            manufacturer = "Nissan5"
        case "Tundra6":
            manufacturer = "Nissan6"
        case _:
            manufacturer = "Unknown"
    
    return manufacturer

def if_elseif_test():
    car = "Patriot"

    if car == "Focus":
        manufacturer = "Ford"
    elif car == "Navigator":
        manufacturer = "Lincoln"
    elif car == "Camry":
        manufacturer = "Toyota"
    elif car == "Civic":
        manufacturer = "Honda"
    elif car == "Patriot":
        manufacturer = "Jeep"
    elif car == "Jetta":
        manufacturer = "VW"
    elif car == "Ceyene":
        manufacturer = "Porsche"
    elif car == "Outback":
        manufacturer = "Subaru"
    elif car == "520i":
        manufacturer = "BMW"
    elif car == "Tundra":
        manufacturer = "Nissan"
    elif car == "Tundra2":
        manufacturer = "Nissan2"
    elif car == "Tundra3":
        manufacturer = "Nissan3"
    elif car == "Tundra4":
        manufacturer = "Nissan4"
    elif car == "Tundra5":
        manufacturer = "Nissan5"
    elif car == "Tundra6":
        manufacturer = "Nissan6"        
    else:
        manufacturer = "Unknown"

    return manufacturer
